code/obj/flock/structure/gnesisturret.dm

```diff
@@ -0,0 +1,163 @@
+//
+/// # Gnesis Turret - Shoots syringes full of coagulated gnesis at poor staffies
+//
+// A vat that slowly generates gnesis over time,
+/obj/flock_structure/gnesisturret
+	name = "spiky fluid vat"
+	desc = "A vat of bubbling teal fluid, covered in hollow spikes."
+	icon_state = "teleblocker-off"
+	flock_id = "Gnesis turret"
+	resourcecost = 300
+	health = 80
+	///maximum volume of coagualted gnesis that can be stored in the tank
+	var/fluid_level_max = 250
+	///how much gnesis is generated per-tick while there is sufficient compute
+	var/fluid_gen_amt = 3
+	///gnesis fluid ID - change this to do exciting things like having a turret that fires QGP
+	var/fluid_gen_type = "flockdrone_fluid"
+	///how much of the stored fluid should be in each shot
+	var/fluid_shot_amt = 20
+	//internals for turret targetting and accuracy
+	var/target = null
+	var/range = 8
+	var/spread = 1
+	var/datum/projectile/syringe/gnesis/current_projectile = null
```

Comment:
probably want this to be `/datum/projectile/syringe/syringe_barbed` so people can't pull them out as easily (and precedent is that antagonists get barbed syringed)

Reply:
https://github.com/stonepillars/goon-flock/pull/333